db/memtable.cc

```diff
@@ -669,6 +669,13 @@ static bool SaveValue(void* arg, const char* entry) {
     if (!s->CheckCallback(seq)) {
       return true;  // to continue to the next seq
     }
+    if (ts_sz > 0) {
+      assert(s->key);
+      SequenceNumber seq_ub = s->key->sequence(/*type=*/nullptr);
+      if (seq_ub < seq) {
+        return true;  // to continue to the next seq
+      }
+    }
```

Comment:
I'm wondering if this functionality could be implemented using the existing `ReadCallback` interface? It seems to be just the thing we need, and it would also eliminate a whole bunch of code changes in `GetContext`'s clients (and also in `LookupKey` above).

Reply:
Good idea. I followed the suggestion and made the PR shorter. The only issue is that we need to pass a read callback to `MultiGet`, which I think can be done in another PR.